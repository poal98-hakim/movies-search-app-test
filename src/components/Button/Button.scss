@use '../../scss/base/colors.scss';

$background-color-button:  #001e2d;

.button{
    width: 6em;
    margin: 10px;
    color: colors.$PRIMARY_COLOR;
    background-color: $background-color-button;
    cursor: pointer;
    border-radius: 5px;
    @media only screen and (max-width:1000px) {
        font-size: 1.1em;
    }
}