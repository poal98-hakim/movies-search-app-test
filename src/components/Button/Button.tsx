import React from "react";
import "./Button.scss";

interface IProps{
  type: string;
  value: string;
}
const Button : React.VFC<IProps> = ({ type, value }) => {
  return <input type={type} className="button" value={value} />
}

export default Button;
