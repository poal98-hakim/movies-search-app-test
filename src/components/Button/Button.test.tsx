import Button from "./Button";
import { render, screen } from '@testing-library/react';

const PROPS = { //mock of Props
  type: "submit",
  value: "Search"
}

describe('Button', () => {
  let props = {...PROPS};

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<Button {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<Button {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads button', () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
  })
  
})
