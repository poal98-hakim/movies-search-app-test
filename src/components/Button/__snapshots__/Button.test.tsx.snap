// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<body>
  <div>
    <input
      class="button"
      type="submit"
      value="Search"
    />
  </div>
  <div>
    <input
      class="button"
      type="submit"
      value="Search"
    />
  </div>
</body>
`;
